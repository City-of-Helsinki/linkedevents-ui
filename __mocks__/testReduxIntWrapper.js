import {IntlProvider, addLocaleData} from 'react-intl'
import {MemoryRouter as Router} from 'react-router'
import {Provider} from 'react-redux'
import React from 'react'

import translations from '../src/i18n'

const ReduxIntWrappeComponent = (store, component, locale = 'fi') => {
    const lang = require('react-intl/locale-data/' + locale)
    addLocaleData(lang)

    return (
        <Provider store={store}>
            <IntlProvider locale={locale} messages={translations[locale]}>
                <Router>
                    {component}
                </Router>
            </IntlProvider>
        </Provider>
    )
}

export default ReduxIntWrappeComponent;
