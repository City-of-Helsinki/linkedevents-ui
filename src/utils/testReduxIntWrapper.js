import { IntlProvider } from 'react-intl'
import { MemoryRouter as Router } from 'react-router'
import { Provider } from 'react-redux'
import React from 'react'
import translations from '../i18n'

export default (store, component, locale = 'fi') => {
  return (
    <Provider store={store}>
      <IntlProvider locale={locale} messages={translations[locale]}>
        <Router>
            {component}
        </Router>
      </IntlProvider>
    </Provider>
  )
}
