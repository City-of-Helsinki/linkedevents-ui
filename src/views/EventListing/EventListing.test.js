import configureStore from 'redux-mock-store'
import React from 'react'
import thunk from 'redux-thunk'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

import testReduxIntWrapper from '../../../__mocks__/testReduxIntWrapper'
import ConnectedEventListing, {EventListing} from './index'
import {mockUserEvents} from '__mocks__/mockData';

const mockStore = configureStore([thunk])
const initialStore = {
    userEvents: {
        isFetching: false,
        fetchComplete: true,
        items: mockUserEvents,
        error: null,
        sortBy: 'last_modified_time',
        sortOrder: 'desc',
        count: 738,
        paginationPage: 0,
    },
    user: {
        id: 'testuser',
        username: 'fooUser',
        provider: 'helsinki',
    },
    events: {
        apiErrorMsg: null,
        isFetching: false,
        fetchComplete: false,
        items: [],
        eventError: null,
        eventsError: null,
    },
    app: {
        flashMsg: null,
        confirmAction: null,
    },
}

describe('EventListing Snapshot', () => {
    let store;

    it('should render view by default', () => {
        const componentProps = {
            fetchUserEvents: jest.fn(),
            login: jest.fn(),
            events: {},
            user: {},
        };
        const wrapper = shallow(<EventListing {...componentProps} />);
        expect(wrapper).toMatchSnapshot();
    })

    it('should render view correctly', () => {
        store = mockStore(initialStore)
        const componentProps = {
            fetchUserEvents: jest.fn(),
            login: jest.fn(),
        } // Props which are added to component
        const componentToTest = <ConnectedEventListing {...componentProps} />
        const wrapper = renderer.create(testReduxIntWrapper(store, componentToTest))

        expect(wrapper).toMatchSnapshot()
    })
})
