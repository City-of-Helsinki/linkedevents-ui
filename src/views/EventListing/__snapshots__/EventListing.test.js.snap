// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListing Snapshot should render view by default 1`] = `
<div
  className="container"
>
  <h1>
    <FormattedMessage
      id="organization-events"
      values={Object {}}
    />
  </h1>
  <p>
    <FormattedMessage
      id="organization-events-description"
      values={Object {}}
    />
  </p>
  <Connect(FilterableEventTable)
    apiErrorMsg=""
    user={Object {}}
  />
</div>
`;

exports[`EventListing Snapshot should render view correctly 1`] = `
<div
  className="container"
>
  <h1>
    <span>
      Tapahtumien hallinta
    </span>
  </h1>
  <p>
    <span>
      Selaa ja muokkaa oman organisaatiosi julkisia tapahtumia sekä luonnoksia. Voit lisätä uuden tapahtuman yläpalkista.
    </span>
  </p>
  <div
    style={
      Object {
        "padding": "0em 2em 0.5em 0em",
      }
    }
  >
    
    <div>
      <table
        className="MuiTable-root-1 event-table"
      >
        <thead>
          <tr
            className="MuiTableRow-root-2 MuiTableRow-head-3"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeHead-9"
            >
              <span
                className="MuiButtonBase-root-20 MuiTableSortLabel-root-15"
                disabled={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span>
                  Otsikko
                </span>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-22 MuiTableSortLabel-icon-17"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeHead-9"
            >
              <span
                className="MuiButtonBase-root-20 MuiTableSortLabel-root-15"
                disabled={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span>
                  Tapahtuma alkaa
                </span>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-22 MuiTableSortLabel-icon-17"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeHead-9"
            >
              <span
                className="MuiButtonBase-root-20 MuiTableSortLabel-root-15"
                disabled={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span>
                  Tapahtuma päättyy
                </span>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-22 MuiTableSortLabel-icon-17"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeHead-9"
            >
              <span
                className="MuiButtonBase-root-20 MuiTableSortLabel-root-15 MuiTableSortLabel-active-16"
                disabled={undefined}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span>
                  Muokattu viimeksi
                </span>
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-22 MuiTableSortLabel-icon-17 MuiTableSortLabel-desc-18"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className="MuiTableRow-root-2"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeBody-10"
            >
              <a
                href="/event/undefined"
                onClick={[Function]}
              >
                TestEvent1
              </a>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeBody-10"
            >
              <span>
                31. tammik. 2018
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeBody-10"
            >
              <span>
                31. tammik. 2018
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeBody-10"
            >
              <span>
                10 kuukautta sitten
              </span>
            </td>
          </tr>
          <tr
            className="MuiTableRow-root-2"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeBody-10"
            >
              <a
                href="/event/undefined"
                onClick={[Function]}
              >
                Pomppu
              </a>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeBody-10"
            >
              <span>
                2. helmik. 2018
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeBody-10"
            >
              <span>
                2. helmik. 2018
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root-7 MuiTableCell-paddingDefault-12 MuiTableCell-typeBody-10"
            >
              <span>
                11 kuukautta sitten
              </span>
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr
            className="MuiTableRow-root-29 MuiTableRow-footer-31"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root-43 MuiTableCell-paddingDefault-48 MuiTableCell-typeFooter-47 MuiTablePagination-root-34"
              colSpan={1000}
            >
              <div
                className="MuiToolbar-root-51 MuiToolbar-gutters-52 MuiTablePagination-toolbar-35"
              >
                <div
                  className="MuiTablePagination-spacer-36"
                />
                <span
                  className="MuiTypography-root-53 MuiTypography-caption-63 MuiTablePagination-caption-37"
                >
                  1-100 / 738
                </span>
                <div
                  className="MuiTablePaginationActions-root-77"
                >
                  <button
                    className="MuiButtonBase-root-87 MuiButtonBase-disabled-88 MuiIconButton-root-78 MuiIconButton-disabled-83"
                    disabled={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role={undefined}
                    tabIndex={-1}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label-84"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-89 MuiIconButton-icon-85"
                        color={undefined}
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    className="MuiButtonBase-root-87 MuiIconButton-root-78"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role={undefined}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label-84"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-89 MuiIconButton-icon-85"
                        color={undefined}
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                    </span>
                    <span
                      className="MuiTouchRipple-root-96"
                    />
                  </button>
                </div>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
